File: frontend/main/src/app/service/slider_service.ts
import { DataDto } from "../types/data-type";

export const sliderService = {
    async newestSlider() {
        const response = await fetch(`/api/slider/newest`, {
            method: 'GET'
        });
        if (!response.ok) throw new Error('Failed to register');
        return response.json() as Promise<DataDto[]>;
    },

    async operatingSystemSlider() {
        const response = await fetch(`/api/slider/system`, {
            method: 'GET'
        });
        if (!response.ok) throw new Error('Failed to register');
        return response.json() as Promise<DataDto[]>;
    },

    async languageSlider() {
        const response = await fetch(`/api/slider/language`, {
            method: 'GET'
        });
        if (!response.ok) throw new Error('Failed to register');
        return response.json() as Promise<DataDto[]>;
    },

    async jobSlider() {
        const response = await fetch(`/api/slider/job`, {
            method: 'GET'
        });
        if (!response.ok) throw new Error('Failed to register');
        return response.json() as Promise<DataDto[]>;
    },

    async aiSlider() {
        const response = await fetch(`/api/slider/ai`, {
            method: 'GET'
        });
        if (!response.ok) throw new Error('Failed to register');
        return response.json() as Promise<DataDto[]>;
    },
}

File: backend/main/service/slider_service.go
package service

import (
	"context"
	"fiber-app/dto"
)

type ISliderService interface {
	NewestSlider(ctx context.Context) ([]dto.DataDto, error)
	OperatingSystemSlider(ctx context.Context) ([]dto.DataDto, error)
	LanguageSlider(ctx context.Context) ([]dto.DataDto, error)
	JobSlider(ctx context.Context) ([]dto.DataDto, error)
	AiSlider(ctx context.Context) ([]dto.DataDto, error)
}